exercises/concept/lasagna-master/lasagna-master.spec.js

```diff
@@ -0,0 +1,214 @@
+  });
+
+  test('uses the default if no customer time was passed', () => {
+    expect(preparationTime(manyLayers())).toBe(24);
+
+    const fewLayers = ['sauce', 'noodles'];
+    expect(preparationTime(fewLayers)).toBe(4);
+  });
+
+  test('works with an empty layers array', () => {
+    expect(preparationTime([])).toBe(0);
+  });
+});
+
+describe('quantities', () => {
+  test('calculates the amounts of noodles and sauce correctly', () => {
+    const fewLayers = ['noodles', 'sauce', 'noodles'];
+    expectObjectsToBeEqual(quantities(fewLayers), { noodles: 100, sauce: 0.2 });
+
+    expectObjectsToBeEqual(quantities(manyLayers()), {
+      noodles: 150,
+      sauce: 0.4,
+    });
+  });
+
+  test('works if there are no noodles or no sauce found in the layers', () => {
+    const noNoodles = ['sauce', 'béchamel', 'sauce', 'meat', 'sauce'];
+    expectObjectsToBeEqual(quantities(noNoodles), { noodles: 0, sauce: 0.6 });
+
+    const noSauce = ['eggplant', 'béchamel', 'noodles', 'béchamel'];
+    expectObjectsToBeEqual(quantities(noSauce), { noodles: 50, sauce: 0 });
+  });
+
+  test('works with an empty layers array', () => {
+    expect(quantities([])).toEqual({ noodles: 0, sauce: 0 });
+  });
+});
+
+describe('addSecretIngredient', () => {
+  test('adds the secret ingredient to the second array', () => {
+    const friendsList = ['sauce', 'noodles', 'béchamel', 'marjoram'];
+    const myList = ['sauce', 'noodles', 'meat', 'tomatoes'];
+    addSecretIngredient(friendsList, myList);
+
+    const expected = ['sauce', 'noodles', 'meat', 'tomatoes', 'marjoram'];
+    expect(myList).toEqual(expected);
+  });
+
+  test('does not modify the first array', () => {
+    const createFriendsList = () => [
+      'noodles',
+      'tomatoes',
+      'sauce',
+      'meat',
+      'mozzarella',
+      'eggplant',
+      'ricotta',
+      'parmesan',
+    ];
+
+    const friendsList = createFriendsList();
+    const myList = ['ricotta', 'béchamel', 'sauce', 'noodles', 'meat'];
+    addSecretIngredient(friendsList, myList);
+
+    expect(friendsList).toEqual(createFriendsList());
+  });
+
+  test('does not return anything', () => {
+    const friendsList = [
+      'sauce',
+      'noodles',
+      'béchamel',
+      'mozzarella',
+      'mustard',
+    ];
+    const myList = ['sauce', 'noodles', 'tomatoes'];
+    expect(addSecretIngredient(friendsList, myList)).toBeUndefined();
+  });
+});
+
+describe('scaleRecipe', () => {
+  test('scales up correctly', () => {
+    const recipe1 = {
+      sauce: 0.5,
+      noodles: 250,
+      meat: 150,
+      tomatoes: 3,
+      onion: 0.5,
+    };
+
+    const expected1 = {
+      sauce: 1.5,
+      noodles: 750,
+      meat: 450,
+      tomatoes: 9,
+      onion: 1.5,
+    };
+
+    expectObjectsToBeEqual(scaleRecipe(recipe1, 6), expected1);
+
+    const recipe2 = {
+      sauce: 0.6,
+      noodles: 300,
+      carrots: 1,
+      mozzarella: 0.5,
+      ricotta: 50,
+      béchamel: 0.1,
```

Comment:
My IDE gives me a warning here:
<img width="523" alt="Screen Shot 2021-08-07 at 10 04 32" src="https://user-images.githubusercontent.com/7852553/128593426-a08ea226-9f1a-4089-a1bc-d8de247d5108.png">

Could it be a problem for some students on some machines? Should the key be wrapped in `''`?

Reply:
The key should be wrapped indeed 🗡️

Reply:
@SleeplessByte I had quotes at some point but prettier/eslint removed them. No matter whether I only quote that one key or all of them. Do you have the same behavior? If yes, any idea how to fix this "globally" instead of ignore rules here?

Reply:
I will check!